models/project.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    @api.onchange('user_ids')
+    def onchange_user_ids(self):
+        for so in self: 
+            for user in so.user_ids: 
+                origin = user.id.origin
+                user_ = self.env['res.users'].search([('id', '=', origin)]) 
+                tasks = self.env['project.task'].search([])
+                assigned_tasks = tasks.filtered(lambda task: origin in task.user_ids.ids and task.sale_order_id)
+                user_.tasks_ids = [(6, 0, assigned_tasks.ids)] 
+                sale_order_ids = set(task.sale_order_id.id for task in assigned_tasks) 
+                user_.stored_sale_orders = [(6, 0, list(sale_order_ids))]
```

Comment:
@iambenkis  have you tried to call `_compute_task_ids` after this function? This way no need to "go in the User view form view to save it manually then access was granted."

Reply:
@sbuhl  The behavior change a bit with the Postgres database. The user table is updated from Postgres, however odoo does not consider the changes made in the DB until I unleash the "save manually" event.

Reply:
@iambenkis  As you change almost everything, can you explain the changes of implementation please?

Reply:
- @sbuhl , well he's a compendium of what I did :
The snippet below represent a callback function that is called when the assignees field is saved :

```python
    def after_save(self):
        users = self.env['res.users'].search([])
        for user in users:
            user._compute_task_ids()   

    def write(self, vals):
        result = super(ProjectTask, self).write(vals)
        self.after_save()
        return result
```

When the `save manually` button is clicked, the after_save e=method is called and updated the res_user table accordingly